vulture/utils.py

```diff
@@ -37,6 +38,17 @@ def read_file(filename):
         raise VultureInputException(err)
 
 
+def return_nodes_after_return(tree):
+    try:
+        for i, node in enumerate(tree.body, 0):
+            if isinstance(node, ast.Return):
+                return tree.body[i+1:]
+    except AttributeError:
+        return []
+    else:
+        return return_nodes_after_return(node)
+
```

Comment:
I don't think either solution is generic enough. I think we need something like this:

```python
def get_ast_lists(tree):
    for node in ast.walk(tree):
        for name, field in ast.iter_fields(node):
            if isinstance(field, list):
                yield field


def search_unreachable_code(tree):
    """
    Iterate over all lists of AST nodes in the given tree. Then for each
    list, check whether there is a Return node that's not at the end of
    the list. If it's not at the end, return an Item that starts at the
    node after the Return node.

    """
    for ast_list in get_ast_lists(tree):
        for index, node in enumerate(ast_list):
            if isinstance(node, ast.Return):
                try:
                    unreachable_node = ast_list[index + 1]
                except IndexError:
                    pass
                else:
                    yield Item(
                        'code after return', 'unreachable code',
                        'filename=todo', unreachable_node.lineno)
```